### PR TITLE
Add opm alpha init/render commands.

### PR DESCRIPTION
**Pull request by joelanford**
**Description of the change:**
The `init` command is a simple CLI used to generate new `olm.package` blobs. It has flags for default channel, icon file, and description file to enable every field to be populated via the CLI.
    
The `render` command converts various opm-related images to a declarative config format. It:
- converts bundle images to `olm.bundle` blobs
- converts databases in sqlite-based index images to declarative config blobs.
- loads and serializes config directories from declarative-config-based index images, inlining any bundle objects that are found.

Note that bundle manifests are NOT rendered into `olm.bundle` blobs when rendering bundle images. The expectation is that the rendered DC can be post-processed to apply any other updates. (e.g. https://github.com/joelanford/declcfg-inline-bundles)

**Motivation for the change:**
Combined, these two commands form the foundation for building new DC-based indexes.

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: internal/action/render.go
Comment (estroz) [inline internal/action/render.go:215]: Is this lossy? Will existing bundle objects contain valid data that `property.MustBuildBundleObjectData([]byte(obj)))` cannot rebuild?

**Review by estroz — CHANGES_REQUESTED** [review]
LGTM overall, mostly nits/questions

File: internal/action/render.go
Comment (awgreene) [inline internal/action/render.go:108]: +1

File: internal/action/render.go
Comment (awgreene) [reply internal/action/render.go:148]: To be fair, the function you linked could be changed in the future and could result in a function call on a nil pointer if this isn't caught.

**Review by awgreene — COMMENTED** [review]
This looks good to me minus the missing newlines and @estroz's suggestions.

File: internal/action/render.go
Comment (joelanford) [reply internal/action/render.go:148]: I copied this code from an existing place in the codebase.

I _generally_ code ([V], error) functions to never return `nil, nil` unless I specifically document that they can. And when I call ([V], error) functions, I make that same assumption. So far, I don't think I've been bitten by nil pointer errors like this.

So :man_shrugging:.. I'll leave it like this in this case because there was already a decision elsewhere to include this check.

File: internal/action/render.go
Comment (joelanford) [reply internal/action/render.go:213]: Yeah, this is in the golang/go SliceTricks for [filtering without allocations](https://github.com/golang/go/wiki/SliceTricks#filtering-without-allocating).

File: internal/action/render.go
Comment (joelanford) [reply internal/action/render.go:215]: If I'm understanding the question, it should not be lossy. In the cases that we care about maintaining these properties (when existing indices are rendered), there are two places to be concerned about:
1. [Converting from sqlite image](https://github.com/operator-framework/operator-registry/blob/c95be3c757ffedf26e8ceb89b6d9462ced1604c6/pkg/api/api_to_model.go#L33).
2. [Rendering an existing DC index image/directory](https://github.com/operator-framework/operator-registry/blob/c95be3c757ffedf26e8ceb89b6d9462ced1604c6/internal/declcfg/load.go#L49-L55).

For bundle images, we intentionally do _not_ populate these because we the `olm.bundle.object` property is a temporary stopgap until the OLM cluster components no longer require the manifests to be sent in the GRPC response. Short term, index maintainers that require these properties would need to post-process the index to include these properties.

File: internal/action/init.go
Comment (joelanford) [reply internal/action/init.go:22]: Let's cover this in a follow-up (I'll add a comment). I have a feeling that we'll need to export more than just the schema names.

File: internal/action/render.go
Comment (joelanford) [reply internal/action/render.go:108]: It isn't clear to me _where_ to put this constant, which is why I added the TODO comment? Any suggestions?

Comment (joelanford) [reply internal/action/render.go:108]: (Maybe this gets rolled into the other follow-up work that will likely happen around exporting some of the DC-related code?)

File: internal/action/render.go
Comment (joelanford) [reply internal/action/render.go:75]: These logs are pretty verbose even in the "happy" case. For example, https://github.com/operator-framework/operator-registry/blame/c95be3c757ffedf26e8ceb89b6d9462ced1604c6/pkg/image/containerdregistry/registry.go#L167 produces a pretty long JSON log string that looks similar to the `docker inspect` output.

Is it reasonable to expect that any messages that need to be surfaced to the user are returned as `error`s rather than logged?

File: cmd/opm/alpha/render/cmd.go
Comment (joelanford) [reply cmd/opm/alpha/render/cmd.go:39]: Similar problem here as in my previous comment. The bundle image parser has some fairly verbose logging in the "happy" case. For example, https://github.com/operator-framework/operator-registry/blob/c95be3c757ffedf26e8ceb89b6d9462ced1604c6/pkg/registry/imageinput.go#L66-L68

Same question: Is it reasonable to expect that any messages that need to be surfaced to the user are returned as errors rather than logged?

Comment (joelanford) [reply cmd/opm/alpha/render/cmd.go:41]: Similar problem here as in my previous comment. The bundle image parser has some fairly verbose logging in the "happy" case. For example, https://github.com/operator-framework/operator-registry/blob/c95be3c757ffedf26e8ceb89b6d9462ced1604c6/pkg/registry/imageinput.go#L66-L68

Same question: Is it reasonable to expect that any messages that need to be surfaced to the user are returned as errors rather than logged?

Comment (estroz) [reply cmd/opm/alpha/render/cmd.go:41]: I think the only case where logs really matter is when debugging. In that case, writing debug logs to stderr works and is idiomatic to my knowledge.

I’m not going to block this PR on logging issues though.

**Review by estroz — APPROVED** [review]
/lgtm

File: cmd/opm/alpha/render/cmd.go
Comment (joelanford) [reply cmd/opm/alpha/render/cmd.go:41]: Yeah, I'm thinking any logging in these libraries should be at the debug level. That way callers aren't forced into this all or nothing situation.

I'll open a new issue against operator-registry so we don't forget about this.

Comment (joelanford) [reply cmd/opm/alpha/render/cmd.go:41]: https://github.com/operator-framework/operator-registry/issues/672

**Review by dinhxuanvu — APPROVED** [review]
/lgtm

**Pull request merged**
